futures-util/src/stream/futures_unordered/node.rs

```diff
@@ -59,32 +73,54 @@ impl<Fut> Node<Fut> {
         }
     }
 
-    // Saftey: The returned `NonNull<Unsafe>` needs to be put into a `Waker`
-    // or `LocalWaker`
-    unsafe fn clone_as_unsafe_wake_without_lifetime(self: &Arc<Node<Fut>>)
-        -> NonNull<dyn UnsafeWake>
-    {
+    /// Returns a local waker for this node without cloning the Arc
+    pub(super) fn local_waker(self: &'a Arc<Node<Fut>>) -> LocalWakerRef<'a> {
```

Comment:
This is now also safe. I created a wrapper struct with a lifetime that stores the `LocalWaker`. Can't return `&LocalWaker` directly from this function because the `LocalWaker` struct has to be stored somewhere. The wrapper struct implements `Deref`, though. So, the usage is nearly identical.